react/features/base/devices/functions.js

```diff
@@ -13,6 +38,71 @@ export function getAudioOutputDeviceId() {
     return JitsiMeetJS.mediaDevices.getAudioOutputDevice();
 }
 
+/**
+ * Finds a device with a label that matches the passed label and returns its id.
+ *
+ * @param {Object} state - The redux state.
+ * @param {string} label - The label.
+ * @returns {string|undefined}
+ */
+export function getDeviceIdByLabel(state: Object, label: string) {
+    const types = [ 'audioInput', 'audioOutput', 'videoInput' ];
+
+    for (const type of types) {
+        const device
+            = state['features/base/devices'].devices[type]
+                .find(d => d.label === label);
+
+        if (device) {
+            return device.deviceId;
+        }
+    }
+}
+
+/**
+ * Returns the devices set in the URL.
+ *
+ * @param {Object} state - The redux state.
+ * @returns {Object|undefined}
+ */
+export function getDevicesFromURL(state: Object) {
+    const urlParams
+        = parseURLParams(state['features/base/connection'].locationURL);
+
+    const audioOutput = urlParams['devices.audioOutput'];
+    const videoInput = urlParams['devices.videoInput'];
+    const audioInput = urlParams['devices.audioInput'];
+
+    if (!audioOutput && !videoInput && !audioInput) {
+        return undefined;
```

Comment:
You don't want to return the empty object or always return an object with the the keys but with falsy values?

Reply:
Why would I want? I'm returning undefined in order to be able to easy detect the use case where there isn't any parameters passed trough the URL.

Reply:
Okay, that intent was not clear.